libs/libGamedata/gameData/CampaignDescription.cpp

```diff
@@ -1,28 +1,42 @@
-// Copyright (C) 2005 - 2021 Settlers Freaks (sf-team at siedler25.org)
+// Copyright (C) 2005 - 2023 Settlers Freaks (sf-team at siedler25.org)
 //
 // SPDX-License-Identifier: GPL-2.0-or-later
 
 #include "CampaignDescription.h"
+#include "helpers/format.hpp"
 #include "lua/CheckedLuaTable.h"
 #include "lua/LuaHelpers.h"
+#include "mygettext/mygettext.h"
 
-CampaignDescription::CampaignDescription() = default;
-
-CampaignDescription::CampaignDescription(CheckedLuaTable luaData, CampaignDescription& campaignDesc)
+CampaignDescription initFromLuaTable(kaguya::LuaTable table)
```

Comment:
This now looks like it can become a regular (and `explicit`) constructor, doesn't it? I mean you create and return a new instance which is basically what constructors are for. Might be even cleaner as you can use an initializer list and only do the validity-checks on the body.

Reply:
Changed to constructor